otter/test/worker/test_launch_server_v1.py

```diff
@@ -2395,13 +2405,48 @@ def test_delete_and_verify_succeeds_if_get_returns_404(self):
         self.treq.delete.return_value = succeed(mock.Mock(code=204))
         self.treq.get.return_value = succeed(mock.Mock(code=404))
 
-        d = delete_and_verify(self.log, 'http://url/', self.request_bag,
-                              'serverId')
+        d = self._delete_and_verify()
         self.assertEqual(len(self.bags), 1)
         self.assertEqual(self.treq.delete.call_count, 1)
         self.assertEqual(self.treq.get.call_count, 1)
         self.successResultOf(d)
 
+    def test_delete_and_verify_limits(self):
+        """
+        :func:`delete_and_verify` limits the number of delete server requests
+        to 1. It also delays response by 1 second
+        """
+        deferreds = [Deferred() for i in range(3)]
+        delete_ds = deferreds[:]
+        self.treq.delete.side_effect = lambda *a, **kw: deferreds.pop(0)
+
+        ret_ds = [delete_and_verify(self.log, 'http://url/',
+                                    self.request_bag, 'serverId', self.clock)
+                  for i in range(3)]
+
+        # no result in any of them and only 1 treq.delete is called
+        for d in ret_ds:
+            self.assertNoResult(d)
+        self.assertEqual(self.treq.delete.call_count, 1)
+
+        # fire first deferred and notice that next treq.delete is still not
+        # called due to delay
+        delete_ds[0].callback(mock.Mock(code=404))
+        self.assertEqual(self.treq.delete.call_count, 1)
+
+        # advance clock and notice next delete called
+        self.clock.advance(1)
+        self.assertEqual(self.treq.delete.call_count, 2)
+        self.successResultOf(ret_ds[0])
```

Comment:
Non-blocking - this may be overkill, wondering if it would make sense to assert that the second deferred has no result yet, even though `treq` was called a second time.